pennylane/templates/subroutines/qsvt.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the QSVT template and qsvt wrapper function.
+"""
+# pylint: disable=too-many-arguments
+import copy
+import pennylane as qml
+from pennylane.queuing import QueuingManager
+from pennylane.ops import BlockEncode, PCPhase
+from pennylane.ops.op_math import adjoint
+from pennylane.operation import AnyWires, Operation
+from pennylane import numpy as np
+
+
+def qsvt(A, angles, wires, convention=None):
+    r"""Performs the
+    `quantum singular value transformation <https://arxiv.org/abs/1806.01838>`__ using
+    :class:`~.BlockEncode` and :class:`~.PCPhase`.
+    See also `A Grand Unification of Quantum Algorithms <https://arxiv.org/pdf/2105.02859.pdf>`__ and
+    :class:`~.QSVT`__.
+
+    Args:
+        A (tensor_like): the general :math:`(n \times m)` matrix to be encoded
+        angles (tensor_like): a list of angles by which to shift to obtain the desired polynomial
+        wires (Iterable): the wires the template acts on
+        convention (string): can be set to "Wx" to convert quantum signal processing angles to QSVT
+
+    **Example**
+
+    >>> dev = qml.device("default.qubit", wires=2)
+    >>> A = [[0.1, 0.2], [0.3, 0.4]]
+    >>> angles = [0.1, 0.2, 0.3]
+
```

Comment:
Might not want to keep spaces between the code as it separates them into different blocks. Output code can have a different block.